cmd/kube-apiserver/app/aggregator.go

```diff
@@ -304,13 +309,78 @@ func apiServicesToRegister(delegateAPIServer genericapiserver.DelegationTarget,
 			continue
 		}
 
+		hash, err := getGroupVersionHash(curr, delegateAPIServer.UnprotectedHandler())
+		if err != nil {
+			klog.Warning(err)
+		}
 		apiService := makeAPIService(schema.GroupVersion{Group: tokens[2], Version: tokens[3]})
 		if apiService == nil {
 			continue
 		}
-		registration.AddAPIServiceToSyncOnStart(apiService)
+		registration.AddAPIServiceToSyncOnStart(apiService, hash)
 		apiServices = append(apiServices, apiService)
 	}
 
 	return apiServices
 }
+
+func getGroupVersionHash(path string, handler http.Handler) (string, error) {
+	// TODO: This can be optimized by using HEAD since we are only interested in the ETag
+	req, err := http.NewRequest("GET", path, nil)
```

Comment:
Is there anything blocking us from using HEAD as the request method as the comment above aspires to? What happens if we change this string to "HEAD"?